kube-client/src/config/file_config.rs

```diff
@@ -66,26 +66,30 @@ pub struct Preferences {
 #[cfg_attr(test, derive(PartialEq, Eq))]
 pub struct NamedExtension {
     /// Name of extension
+    #[serde(default)]
     pub name: String,
     /// Additional information for extenders so that reads and writes don't clobber unknown fields
     pub extension: serde_json::Value,
 }
 
 /// NamedCluster associates name with cluster.
-#[derive(Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
 #[cfg_attr(test, derive(PartialEq, Eq))]
 pub struct NamedCluster {
     /// Name of cluster
+    #[serde(default)]
     pub name: String,
     /// Information about how to communicate with a kubernetes cluster
-    pub cluster: Cluster,
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
Shouldn't this be:

```
#[serde(default, skip_serializing_if = "Option::is_none")]
```

Reply:
Yeah you're right it doesn't make a difference, unfortunately I didnt find anything in the docs about this. But the expanded macro code handles options differently and implicitly works like if there was a `default` applied to it. 👍